sparkl_cli/test/test_password.py

```diff
@@ -0,0 +1,218 @@
+
+    Registers and deletes an additional test user for each test.
+
+    Tests can use this user to change its password. The setup also \
+    hands down an open connection for tests use.
+
+    It depends on the central ``connect_sparkl_setup`` \
+    test setup method.
+
+    """
+
+    test_alias = "setup_test_alias"
+    test_user = "yatu@localhost"
+    test_password = "yapassword"
+
+    try:
+        sparkl(
+            "connect",
+            connect_sparkl_setup["TEST_SSE"],
+            alias=test_alias)
+
+        register_user(
+            connect_sparkl_setup["TEST_SSE"],
+            test_user,
+            test_password)
+
+        yield test_alias, test_user, test_password
+
+    finally:
+        sparkl("close", alias=test_alias)
+
+        delete_user(
+            connect_sparkl_setup["TEST_SSE"],
+            connect_sparkl_setup["TEST_ADMIN_USER"],
+            connect_sparkl_setup["TEST_ADMIN_PASS"],
+            test_user)
+
+
+def test_change_non_admin(setup_test_password):
+    alias, user, password = setup_test_password
+    result = sparkl(
+        "login",
+        user,
+        password,
+        alias=alias)
+    assert result["tag"] == "user"
+    assert result["attr"]["name"] == user
+    assert not result["attr"]["administrator"]
+
+    with pytest.raises(CliException):
+        sparkl("password",
+               password,
+               "new_password",
+               alias=alias)
+
+
+def test_change_no_such_user(setup_test_password_admin, setup_test_password):
+    alias, _user, password = setup_test_password
+    admin_alias, admin_user = setup_test_password_admin
+
+    result = sparkl(
+        "user",
+        alias=admin_alias)
+    assert result["tag"] == "user"
+    assert result["attr"]["name"] == admin_user
+    assert result["attr"]["administrator"]
+
+    with pytest.raises(CliException):
+        sparkl("password",
+               password,
+               "new_password",
+               user="no_such_user",
+               alias=admin_alias)
+
+
+def test_change_password(setup_test_password_admin, setup_test_password):
+    alias, user, password = setup_test_password
+    new_password = "another_password"
+
+    admin_alias, admin_user = setup_test_password_admin
+
+    result = sparkl(
+        "user",
+        alias=admin_alias)
+    assert result["tag"] == "user"
+    assert result["attr"]["name"] == admin_user
+    assert result["attr"]["administrator"]
+
+    result = sparkl(
+        "password",
+        password,
+        new_password,
+        alias=admin_alias, user=user)
+
+    assert result["tag"] == "ok"
+    result = sparkl("login", user, new_password, alias=alias)
+    assert result["attr"]["name"] == user
+
+
+@mock.patch("getpass.getpass")
+def test_change_prompt(
+        mock_get_pass, setup_test_password_admin, setup_test_password):
+    """
+    Mocks out the getpass method used to prompt for a password if \
+    not provided.
```

Comment:
I found the doctest misleading, as if the main point of this function were to mock rather than testing.

```suggestion
    Tests changing the password without providing the passwords as arguments.

    As not providing the passwords as arguments causes the command to prompt,
    it mocks out the getpass method used to prompt for a password if \
    not provided.
```